library/fsio.c

```diff
@@ -0,0 +1,417 @@
+/*
+ *  Serialization based implementation of filesystem IO functions.
+ *
+ *  Copyright (C) 2017, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "mbedtls/config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+
+#if defined(MBEDTLS_FS_IO)
+
+#include <string.h>
+#include <assert.h>
+#include "mbedtls/fsio.h"
+
+#if !defined(MBEDTLS_FS_IO_ALT)
+
+#if !defined(MBEDTLS_PLATFORM_NO_STD_FUNCTIONS)
+/**
+ * \brief          Open file and disable buffering.
+ *
+ * \param path     File path
+ * \param mode     Open mode
+ *
+ * \return         Pointer to mbedtls_file_t on success or NULL on failure.
+ */
+mbedtls_file_t * mbedtls_fopen( const char *path, const char *mode )
+{
+    mbedtls_file_t * f = fopen( path, mode );
+
+    if( f && ( setvbuf( f, NULL, _IONBF, 0 ) != 0 ) )
```

Comment:
Its not a validity check since ```f``` returned by ```fopen``` is returned as is. The check is to make sure ```setvbuf``` can be called. The idea of this wrapper is to add call to ```setvbuf``` with ```fopen```.